From eabcc59b9aaab47d5f6b3b692b5b477d29445a05 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Stefan=20D=C3=B6singer?= <stefan@codeweavers.com>
Date: Mon, 5 Aug 2013 13:07:42 +0200
Subject: [PATCH 108/127] wined3d: Don't reset the query state if it doesn't
 have a ctx
Reply-To: wine-devel <wine-devel@winehq.org>

The CS recreates occlusion queries and changes the context variable.
This leads to incorrect query results.

TODO: Before sending, find out why the line was there in the first place
and if it is needed with the multithreaded and/or singlethreaded CS.
---
 dlls/wined3d/query.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/dlls/wined3d/query.c b/dlls/wined3d/query.c
index bc4fc16..34e17da 100644
--- a/dlls/wined3d/query.c
+++ b/dlls/wined3d/query.c
@@ -308,9 +308,6 @@ static HRESULT wined3d_occlusion_query_ops_get_data(struct wined3d_query *query,
     struct wined3d_occlusion_query *oq = query->extendedData;
     DWORD *data = pData;
 
-    if (!oq->context)
-        query->state = QUERY_CREATED;
-
     if (query->state == QUERY_CREATED)
     {
         /* D3D allows GetData on a new query, OpenGL doesn't. So just invent the data ourselves */
-- 
1.8.1.5

